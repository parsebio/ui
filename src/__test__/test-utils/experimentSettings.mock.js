import _ from 'lodash';

import initialState from 'redux/reducers/experimentSettings/initialState';
import { analysisTools } from 'utils/constants';

const sampleifiedConfig = (sampleIds, configToReplicate) => {
  const result = sampleIds.reduce(
    (acum, sampleId) => {
      // eslint-disable-next-line no-param-reassign
      acum[sampleId] = configToReplicate;
      return acum;
    },
    {},
  );

  return result;
};

const generateProcessingConfigMock = (sampleIds) => ({
  classifier: {
    ...sampleifiedConfig(sampleIds, {
      auto: true,
      enabled: true,
      prefiltered: false,
      filterSettings: {
        FDR: 0.1,
      },
    }),
  },
  cellSizeDistribution: {
    ...sampleifiedConfig(sampleIds, {
      auto: true,
      enabled: true,
      filterSettings: {
        minCellSize: 10800,
        binStep: 200,
      },
    }),
  },
  mitochondrialContent: {
    ...sampleifiedConfig(sampleIds, {
      auto: true,
      enabled: true,
      filterSettings: {
        method: 'absoluteThreshold',
        methodSettings: {
          absoluteThreshold: {
            maxFraction: 0.1,
            binStep: 200,
          },
        },
      },
    }),
  },
  numGenesVsNumUmis: {
    ...sampleifiedConfig(sampleIds, {
      auto: true,
      enabled: true,
      filterSettings: {
        regressionType: 'gam',
        regressionTypeSettings: {
          gam: {
            'p.level': 0.00009,
          },
        },
      },
    }),
  },
  doubletScores: {
    ...sampleifiedConfig(sampleIds, {
      auto: true,
      enabled: true,
      filterSettings: {
        probabilityThreshold: 0.2,
        binStep: 0.05,
      },
    }),
  },
  dataIntegration: {
    enabled: true,
    analysisTool: analysisTools.SEURAT,
    dataIntegration: {
      method: 'harmony',
      methodSettings: {
        harmony: {
          numGenes: 2000,
          normalisation: 'logNormalize',
        },
        seuratv4: {
          numGenes: 2000,
          normalisation: 'logNormalize',
        },
        fastmnn: {
          numGenes: 2000,
          normalisation: 'logNormalize',
        },
        unisample: {
          numGenes: 2000,
          normalisation: 'logNormalize',
        },
      },
    },
    dimensionalityReduction: {
      method: 'rpca',
      numPCs: 30,
      excludeGeneCategories: [],
    },
  },
  configureEmbedding: {
    enabled: true,
    embeddingSettings: {
      method: 'umap',
      useSaved: false,
      methodSettings: {
        umap: {
          minimumDistance: 0.1,
          distanceMetric: 'euclidean',
        },
        tsne: {
          perplexity: 30,
          learningRate: 200,
        },
      },
    },
    clusteringSettings: {
      method: 'louvain',
      methodSettings: {
        louvain: {
          resolution: 0.5,
        },
      },
    },
  },
});

const generateExperimentSettingsMock = (sampleIds) => {
  const mockedProcessingConfig = {
    ...initialState.processing,
    ...generateProcessingConfigMock(sampleIds),
  };

  return {
    ...initialState,
    processing: mockedProcessingConfig,
    originalProcessing: _.cloneDeep(mockedProcessingConfig),
    info: {
      ...initialState.info,
      pipelineVersion: 1,
    },
  };
};

export default generateExperimentSettingsMock;
