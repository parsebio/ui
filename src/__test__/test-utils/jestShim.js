/**
 * This file is required for jest to work with '@cfaester/enzyme-adapter-react-18'
 * module.
 */
import {
  TextDecoder, TextEncoder,
} from 'util';

global.TextEncoder = TextEncoder;
global.TextDecoder = TextDecoder;
