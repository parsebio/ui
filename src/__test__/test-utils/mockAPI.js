import _ from 'lodash';
import backendStatusData from '__test__/data/backend_status.json';
import processingConfigData from '__test__/data/processing_config.json';
import mockDemoExperiments from '__test__/test-utils/mockData/mockDemoExperiments.json';
import { mockSecondaryAnalyses } from '__test__/data/secondaryAnalyses/secondary_analyses';
import mockSecondaryAnalysisStatusDefault from '__test__/data/secondaryAnalyses/secondary_analysis_status_default.json';
import fake from '__test__/test-utils/constants';
import mockAnalysisFiles from '__test__/data/secondaryAnalyses/secondary_analysis_files';
import downloadFromS3 from 'utils/work/downloadFromS3';

import {
  responseData,
} from '__test__/test-utils/mockData';
import { analysisTools } from 'const';

const cellSetsData = require('__test__/data/cell_sets.json');

const setupDownloadCellSetsFromS3Mock = (customCellSets = cellSetsData) => {
  downloadFromS3.mockImplementation((resource, signedUrl) => {
    // eslint-disable-next-line global-require
    if (signedUrl === 'mock-cellsets-signed-url') {
      return Promise.resolve(customCellSets);
    }
    return Promise.reject(new Error('Invalid signed URL'));
  });
};

const promiseResponse = (response, options = {}) => (
  Promise.resolve(new Response(response, options)));

const statusResponse = (code, body) => (
  Promise.resolve({
    status: code,
    body: JSON.stringify(body),
  })
);

const delayedResponse = (response, delay = 10000, options = {}) => new Promise((resolve) => {
  setTimeout(() => {
    resolve(new Response(response, options));
  }, delay);
});

const workerDataResult = (data) => Promise.resolve(_.cloneDeep(data));

const fetchWorkMock = (mockedResults) => ((experimentId, body) => {
  if (typeof mockedResults[body.name] === 'function') {
    return mockedResults[body.name]();
  }
  return workerDataResult(mockedResults[experimentId]);
});

const generateDefaultMockAPIResponses = (projectId, analysisTool = analysisTools.SEURAT) => ({
  [`experiments/${projectId}$`]: () => promiseResponse(
    JSON.stringify(responseData.experiments.find(({ id }) => id === projectId)),
  ),
  [`/v2/experiments/${projectId}$`]: () => promiseResponse(
    JSON.stringify(responseData.experiments.find(({ id }) => id === projectId)),
  ),
  [`experiments/${projectId}/processingConfig$`]: () => {
    const processingConfigDataClone = _.cloneDeep(processingConfigData);
    processingConfigDataClone.processingConfig.dataIntegration.analysisTool = analysisTool;

    return promiseResponse(
      JSON.stringify(processingConfigDataClone),
    );
  },
  [`experiments/${projectId}/cellSets$`]: () => {
    setupDownloadCellSetsFromS3Mock();
    return promiseResponse(
      JSON.stringify('mock-cellsets-signed-url'),
    );
  },
  [`experiments/${projectId}/backendStatus$`]: () => promiseResponse(
    JSON.stringify(backendStatusData),
  ),
  experiments$: () => promiseResponse(
    JSON.stringify(responseData.experiments),
  ),
  [`experiments/${projectId}/samples$`]: () => promiseResponse(
    JSON.stringify(responseData.samples[0]),
  ),
  '/v2/experiments/examples$': () => promiseResponse(
    JSON.stringify(mockDemoExperiments),
  ),
  'experiments/clone$': () => promiseResponse(
    JSON.stringify(fake.CLONED_EXPERIMENT_ID),
  ),
  [`/v2/workRequest/${projectId}`]: () => statusResponse(200, 'OK'),
  '/v2/secondaryAnalysis$': () => {
    const mockedMainSecondary = _.cloneDeep(mockSecondaryAnalyses[1]);
    mockedMainSecondary.id = projectId;

    return promiseResponse(
      JSON.stringify([
        mockedMainSecondary,
        ...mockSecondaryAnalyses,
      ]),
    );
  },
  [`/v2/secondaryAnalysis/${projectId}/executionStatus`]: () => promiseResponse(
    JSON.stringify(mockSecondaryAnalysisStatusDefault),
  ),
  [`/v2/secondaryAnalysis/${projectId}/files`]: (req) => {
    if (req.method === 'GET') {
      return promiseResponse(
        JSON.stringify(mockAnalysisFiles),
      );
    }

    if (req.method === 'POST') {
      return promiseResponse(
        JSON.stringify({ fileId: 'mockFileId' }),
      );
    }
  },
  [`/v2/projects/${projectId}/upload/.*/part/\\d+/signedUrl$`]: (req) => {
    const partNumber = req.url.match(/part\/(\d+)\/signedUrl/)[1];
    return promiseResponse(JSON.stringify(`mockSignedUrl${partNumber}`));
  },
  '/v2/cliUpload/token$': () => promiseResponse(JSON.stringify('token')),
});

const mockAPI = (apiMapping) => (req) => {
  const path = req.url;

  const key = _.find(Object.keys(apiMapping), (matcherStr) => new RegExp(matcherStr).test(path));

  if (!key) {
    return statusResponse({
      status: 404,
      body: `Path ${path} is undefined`,
    });
  }

  return apiMapping[key](req);
};

export default mockAPI;
export {
  generateDefaultMockAPIResponses,
  promiseResponse,
  statusResponse,
  delayedResponse,
  workerDataResult,
  fetchWorkMock,
  setupDownloadCellSetsFromS3Mock,
};
