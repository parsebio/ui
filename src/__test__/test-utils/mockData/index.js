import _ from 'lodash';
import { analysisTools } from 'utils/constants';
import generateMockSamples from './generateMockSamples';
import generateMockExperiments from './generateMockExperiments';
import generateMockProcessingConfig from './generateMockProcessingConfig';

// This file is a work in progress to generate mock data for the endpoints of the test.
// The generated data should be usable as response data to be sent via the mocked API in mockAPI.js
// The module should also expose the generated data that can be imported and used easily in tests

const responseData = {};

responseData.experiments = generateMockExperiments(2);

responseData.samples = [generateMockSamples(
  responseData.experiments[0],
  3,
)];

responseData.processingConfigSeurat = generateMockProcessingConfig(analysisTools.SEURAT, 3);
responseData.processingConfigScanpy = generateMockProcessingConfig(analysisTools.SCANPY, 3);

// Add samples to first experiment
const samples = responseData.samples[0];
const sampleIds = _.map(samples, 'id');
responseData.experiments[0].samplesOrder = sampleIds;

const { experiments, processingConfigSeurat, processingConfigScanpy } = responseData;

export {
  experiments,
  samples,
  processingConfigSeurat,
  processingConfigScanpy,
  responseData,
};
