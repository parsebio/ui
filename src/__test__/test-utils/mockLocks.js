const setupNavigatorLocks = () => {
  const mockLockRequest = jest.fn((lock, func) => func());
  global.navigator.locks = {
    request: mockLockRequest,
  };

  return mockLockRequest;
};

const teardownNavigatorLocks = () => {
  delete global.navigator.locks;
};

export { setupNavigatorLocks, teardownNavigatorLocks };
