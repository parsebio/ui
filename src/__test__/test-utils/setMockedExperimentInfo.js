import { updateExperimentInfo } from 'redux/actions/experimentSettings';

const setMockedExperimentInfo = async (experimentId, store) => {
  await store.dispatch(
    updateExperimentInfo({
      experimentId,
      experimentName: 'mockedName',
      sampleIds: ['1', '2'],
      pipelineVersion: 'v2',
      accessRole: 'owner',
    }),
  );
};

export default setMockedExperimentInfo;
