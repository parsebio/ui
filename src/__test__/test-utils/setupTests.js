import '@testing-library/jest-dom/extend-expect';
import preloadAll from 'jest-next-dynamic';

import Enzyme from 'enzyme';
import Adapter from '@cfaester/enzyme-adapter-react-18';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('localforage');
jest.mock('utils/pushNotificationMessage');

// This is needed, because the unit tests don't register the Vitessce imports
jest.mock('components/data-exploration/DynamicVitessceWrappers', () => ({
  Scatterplot: () => 'Mocked Scatterplot',
  Heatmap: () => 'Mocked Heatmap',
}));

jest.mock('utils/work/downloadFromS3');

// The updated versions of react - dnd seem to not have a compatible version
// of the testing libraries anymore, so trying anything with the old way of setting up tests
// (with wrapWithTestBackend from react-dnd-test-utils or similar options) breaks.

jest.mock('react-dnd', () => {
  const mockBackend = { /* add methods as needed for your tests */ };
  const mockManager = {
    getBackend: jest.fn(() => mockBackend),
    getMonitor: jest.fn(() => ({
      subscribeToStateChange: jest.fn(),
      subscribeToOffsetChange: jest.fn(),
      isDragging: jest.fn(() => false),
      getItemType: jest.fn(),
      getItem: jest.fn(),
      canDragSource: jest.fn(),
      canDropOnTarget: jest.fn(),
      getDropResult: jest.fn(),
      didDrop: jest.fn(),
      getInitialClientOffset: jest.fn(),
      getInitialSourceClientOffset: jest.fn(),
      getSourceClientOffset: jest.fn(),
      getClientOffset: jest.fn(),
      getDifferenceFromInitialOffset: jest.fn(),
      isOverTarget: jest.fn(() => false),
      getTargetIds: jest.fn(),
    })),
    getActions: jest.fn(() => ({
      beginDrag: jest.fn(),
      publishDragSource: jest.fn(),
      hover: jest.fn(),
      drop: jest.fn(),
      endDrag: jest.fn(),
    })),
    getRegistry: jest.fn(() => ({
      addSource: jest.fn(),
      addTarget: jest.fn(),
      removeSource: jest.fn(),
      removeTarget: jest.fn(),
    })),
  };
  return {
    useDrag: () => [{ isDragging: false, handlerId: 'mock', draggedItem: null }, jest.fn(), jest.fn()],
    useDrop: () => [{ isOver: false, canDrop: false, handlerId: 'mock' }, jest.fn()],
    useDragDropManager: () => mockManager,
    DndProvider: ({ children }) => children,
  };
});

jest.mock('react-dnd-multi-backend', () => ({
  TouchTransition: {},
  HTML5toTouch: {},
  DndProvider: ({ children }) => children,
  createTransition: jest.fn(),
  default: {},
}));

jest.mock('rdndmb-html5-to-touch', () => ({
  default: {},
  HTML5toTouch: {},
}));

beforeAll(async () => {
  // Add stuff that needs to run once, before all tests
  await preloadAll();
});

beforeEach(async () => {
  // Add stuff that needs to run before each test
});

// setImmediate is not available in JSDOM, only on node
// https://github.com/prisma/prisma/issues/8558#issuecomment-1006100001
global.setImmediate = global.setImmediate || ((fn, ...args) => global.setTimeout(fn, 0, ...args));
