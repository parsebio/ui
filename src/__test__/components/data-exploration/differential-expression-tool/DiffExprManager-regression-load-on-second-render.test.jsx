import React from 'react';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import waitForActions from 'redux-mock-store-await-actions';

import DiffExprManager from 'components/data-exploration/differential-expression-tool/DiffExprManager';
import DiffExprCompute from 'components/data-exploration/differential-expression-tool/DiffExprCompute';
import DiffExprResults from 'components/data-exploration/differential-expression-tool/DiffExprResults';

import { mockCellSets } from '__test__/test-utils/cellSets.mock';

import { DIFF_EXPR_LOADING, DIFF_EXPR_LOADED, DIFF_EXPR_ORDERING_SET } from 'redux/actionTypes/differentialExpression';

jest.mock('utils/work/fetchWork', () => (
  jest.fn(() => new Promise((resolve) => resolve({
    data: {
      p_val: [1.496, 2.496, 3.496, 4.496, 5.496],
      p_val_adj: [1.647, 2.647, 3.647, 4.647, 5.647],
      logFC: [-1.427, -2.427, -3.427, -4.427, -5.427],
      gene_names: ['A', 'B', 'C', 'D', 'E'],
      Gene: ['EASAD0', 'ENASD23', 'EN34S', 'ENSD33', 'ENASD233'],

      auc: ['0.1', '0.2', '0.3', '0.4', '0.5'],
      pct_1: ['100', '90', '80', '70', '60'],
      pct_2: ['100', '90', '80', '70', '60'],
    },
    total: 500,
  })))
));

const mockStore = configureMockStore([thunk]);

const experimentId = '1234';

const backendStatus = {
  [experimentId]: {
    status: {
      pipeline: {
        startDate: '2021-01-01T00:00:00',
      },
    },
  },
};

const storeState = {
  genes: {
    selected: [],
  },
  cellInfo: {
    focus: {
      store: 'genes',
      key: 'C',
    },
  },
  cellSets: mockCellSets,
  differentialExpression: {
    properties: {
      data: [
        {
          p_val: 1.496, p_val_adj: 1.647, logFC: -1.427, gene_names: 'A', auc: '0.1', pct_1: '100', pct_2: '100',
        },
        {
          p_val: 2.496, p_val_adj: 2.647, logFC: -2.427, gene_names: 'B', auc: '0.2', pct_1: '90', pct_2: '90',
        },
        {
          p_val: 3.496, p_val_adj: 3.647, logFC: -3.427, gene_names: 'C', auc: '0.3', pct_1: '80', pct_2: '80',
        },
        {
          p_val: 4.496, p_val_adj: 4.647, logFC: -4.427, gene_names: 'D', auc: '0.4', pct_1: '70', pct_2: '70',
        },
        {
          p_val: 5.496, p_val_adj: 5.647, logFC: -5.427, gene_names: 'E', auc: '0.5', pct_1: '60', pct_2: '60',
        },
      ],
      loading: false,
      error: false,
      total: 5,
    },
    comparison: {
      type: 'between',
      group: {
        between: {
          cellSet: 'cluster-a',
          compareWith: 'cluster-b',
          basis: 'scratchpad-a',
        },
      },
      advancedFilters: [],
    },
  },
  backendStatus,
};

let store = null;

describe('DiffExprManager regression test - diff exp would not reload after `go back` was hit and a new cluster selected', () => {
  beforeEach(() => {
    store = mockStore(storeState);
  });

  it('on click of compute with changed parameters, DiffExprManager calls the results view and dispatches the appropriate actions', async () => {
    const component = mount(
      <Provider store={store}>
        <DiffExprManager experimentId={experimentId} view='compute' width={100} height={200} />
      </Provider>,
    );

    const comparisonGroup = { cellSet: 'louvain/cluster-1', compareWith: 'rest', basis: 'all' };
    act(() => {
      component.find(DiffExprCompute).props().onCompute('between', comparisonGroup);
    });
    component.update();

    expect(component.find(DiffExprResults).length).toEqual(1);
    expect(component.find(DiffExprCompute).length).toEqual(0);

    await waitForActions(store, [DIFF_EXPR_ORDERING_SET, DIFF_EXPR_LOADING, DIFF_EXPR_LOADED]);

    expect(store.getActions().length).toEqual(3);
    expect(store.getActions()[1]).toMatchSnapshot();
    expect(store.getActions()[2]).toMatchSnapshot();
  });

  it('if we then go back and change the parameters again, the new differential expression data should be loading', async () => {
    const component = mount(
      <Provider store={store}>
        <DiffExprManager experimentId={experimentId} view='compute' width={100} height={200} />
      </Provider>,
    );

    // Choose a cluster and hit compute.
    let comparisonGroup = { cellSet: 'louvain/cluster-2', compareWith: 'rest', basis: 'all' };
    act(() => {
      component.find(DiffExprCompute).props().onCompute('between', comparisonGroup);
    });
    component.update();

    expect(component.find(DiffExprResults).length).toEqual(1);
    expect(component.find(DiffExprCompute).length).toEqual(0);

    // Ensure load diff exp was called.
    await waitForActions(store, [DIFF_EXPR_ORDERING_SET, DIFF_EXPR_LOADING, DIFF_EXPR_LOADED]);

    expect(store.getActions().length).toEqual(3);
    expect(store.getActions()[1]).toMatchSnapshot();
    expect(store.getActions()[2]).toMatchSnapshot();

    // Go back.
    act(() => {
      component.find(DiffExprResults).props().onGoBack();
    });
    component.update();

    // Ensure original is still rendered correctly.
    expect(component.find(DiffExprResults).length).toEqual(0);
    expect(component.find(DiffExprCompute).length).toEqual(1);

    // Choose another cell set.
    comparisonGroup = { cellSet: 'louvain/cluster-3', compareWith: 'rest', basis: 'all' };
    act(() => {
      component.find(DiffExprCompute).props().onCompute('between', comparisonGroup);
    });
    component.update();

    // Ensure load diff exp was called again.
    await waitForActions(store, [DIFF_EXPR_ORDERING_SET, DIFF_EXPR_LOADING, DIFF_EXPR_LOADED]);

    expect(store.getActions().length).toEqual(6);
    expect(store.getActions()[2]).toMatchSnapshot();
    expect(store.getActions()[3]).toMatchSnapshot();
  });
});
