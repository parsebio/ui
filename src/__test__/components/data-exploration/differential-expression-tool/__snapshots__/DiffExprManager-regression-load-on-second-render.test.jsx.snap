// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiffExprManager regression test - diff exp would not reload after \`go back\` was hit and a new cluster selected if we then go back and change the parameters again, the new differential expression data should be loading 1`] = `
{
  "payload": {
    "advancedFilters": [],
    "experimentId": "1234",
  },
  "type": "diffExpr/loading",
}
`;

exports[`DiffExprManager regression test - diff exp would not reload after \`go back\` was hit and a new cluster selected if we then go back and change the parameters again, the new differential expression data should be loading 2`] = `
{
  "payload": {
    "comparisonGroup": {
      "basis": "scratchpad-a",
      "cellSet": "cluster-a",
      "compareWith": "cluster-b",
    },
    "comparisonType": "between",
    "data": [
      {
        "Gene": "EASAD0",
        "auc": "0.1",
        "gene_names": "A",
        "logFC": -1.427,
        "p_val": 1.496,
        "p_val_adj": 1.647,
        "pct_1": "100",
        "pct_2": "100",
      },
      {
        "Gene": "ENASD23",
        "auc": "0.2",
        "gene_names": "B",
        "logFC": -2.427,
        "p_val": 2.496,
        "p_val_adj": 2.647,
        "pct_1": "90",
        "pct_2": "90",
      },
      {
        "Gene": "EN34S",
        "auc": "0.3",
        "gene_names": "C",
        "logFC": -3.427,
        "p_val": 3.496,
        "p_val_adj": 3.647,
        "pct_1": "80",
        "pct_2": "80",
      },
      {
        "Gene": "ENSD33",
        "auc": "0.4",
        "gene_names": "D",
        "logFC": -4.427,
        "p_val": 4.496,
        "p_val_adj": 4.647,
        "pct_1": "70",
        "pct_2": "70",
      },
      {
        "Gene": "ENASD233",
        "auc": "0.5",
        "gene_names": "E",
        "logFC": -5.427,
        "p_val": 5.496,
        "p_val_adj": 5.647,
        "pct_1": "60",
        "pct_2": "60",
      },
    ],
    "experimentId": "1234",
    "total": 500,
  },
  "type": "diffExpr/loaded",
}
`;

exports[`DiffExprManager regression test - diff exp would not reload after \`go back\` was hit and a new cluster selected if we then go back and change the parameters again, the new differential expression data should be loading 3`] = `
{
  "payload": {
    "comparisonGroup": {
      "basis": "scratchpad-a",
      "cellSet": "cluster-a",
      "compareWith": "cluster-b",
    },
    "comparisonType": "between",
    "data": [
      {
        "Gene": "EASAD0",
        "auc": "0.1",
        "gene_names": "A",
        "logFC": -1.427,
        "p_val": 1.496,
        "p_val_adj": 1.647,
        "pct_1": "100",
        "pct_2": "100",
      },
      {
        "Gene": "ENASD23",
        "auc": "0.2",
        "gene_names": "B",
        "logFC": -2.427,
        "p_val": 2.496,
        "p_val_adj": 2.647,
        "pct_1": "90",
        "pct_2": "90",
      },
      {
        "Gene": "EN34S",
        "auc": "0.3",
        "gene_names": "C",
        "logFC": -3.427,
        "p_val": 3.496,
        "p_val_adj": 3.647,
        "pct_1": "80",
        "pct_2": "80",
      },
      {
        "Gene": "ENSD33",
        "auc": "0.4",
        "gene_names": "D",
        "logFC": -4.427,
        "p_val": 4.496,
        "p_val_adj": 4.647,
        "pct_1": "70",
        "pct_2": "70",
      },
      {
        "Gene": "ENASD233",
        "auc": "0.5",
        "gene_names": "E",
        "logFC": -5.427,
        "p_val": 5.496,
        "p_val_adj": 5.647,
        "pct_1": "60",
        "pct_2": "60",
      },
    ],
    "experimentId": "1234",
    "total": 500,
  },
  "type": "diffExpr/loaded",
}
`;

exports[`DiffExprManager regression test - diff exp would not reload after \`go back\` was hit and a new cluster selected if we then go back and change the parameters again, the new differential expression data should be loading 4`] = `
{
  "payload": {
    "orderBy": "logFC",
    "orderDirection": "DESC",
  },
  "type": "diffExpr/ordering",
}
`;

exports[`DiffExprManager regression test - diff exp would not reload after \`go back\` was hit and a new cluster selected on click of compute with changed parameters, DiffExprManager calls the results view and dispatches the appropriate actions 1`] = `
{
  "payload": {
    "advancedFilters": [],
    "experimentId": "1234",
  },
  "type": "diffExpr/loading",
}
`;

exports[`DiffExprManager regression test - diff exp would not reload after \`go back\` was hit and a new cluster selected on click of compute with changed parameters, DiffExprManager calls the results view and dispatches the appropriate actions 2`] = `
{
  "payload": {
    "comparisonGroup": {
      "basis": "scratchpad-a",
      "cellSet": "cluster-a",
      "compareWith": "cluster-b",
    },
    "comparisonType": "between",
    "data": [
      {
        "Gene": "EASAD0",
        "auc": "0.1",
        "gene_names": "A",
        "logFC": -1.427,
        "p_val": 1.496,
        "p_val_adj": 1.647,
        "pct_1": "100",
        "pct_2": "100",
      },
      {
        "Gene": "ENASD23",
        "auc": "0.2",
        "gene_names": "B",
        "logFC": -2.427,
        "p_val": 2.496,
        "p_val_adj": 2.647,
        "pct_1": "90",
        "pct_2": "90",
      },
      {
        "Gene": "EN34S",
        "auc": "0.3",
        "gene_names": "C",
        "logFC": -3.427,
        "p_val": 3.496,
        "p_val_adj": 3.647,
        "pct_1": "80",
        "pct_2": "80",
      },
      {
        "Gene": "ENSD33",
        "auc": "0.4",
        "gene_names": "D",
        "logFC": -4.427,
        "p_val": 4.496,
        "p_val_adj": 4.647,
        "pct_1": "70",
        "pct_2": "70",
      },
      {
        "Gene": "ENASD233",
        "auc": "0.5",
        "gene_names": "E",
        "logFC": -5.427,
        "p_val": 5.496,
        "p_val_adj": 5.647,
        "pct_1": "60",
        "pct_2": "60",
      },
    ],
    "experimentId": "1234",
    "total": 500,
  },
  "type": "diffExpr/loaded",
}
`;
